specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2023-03-01-preview/examples/Caches_CreateOrUpdate.json

```diff
@@ -0,0 +1,327 @@
+{
+  "parameters": {
+    "resourceGroupName": "scgroup",
+    "cacheName": "sc1",
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "api-version": "2023-03-01-preview",
+    "cache": {
+      "tags": {
+        "Dept": "Contoso"
+      },
+      "location": "westus",
+      "identity": {
+        "type": "UserAssigned",
+        "userAssignedIdentities": {
+          "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/scgroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/identity1": {}
+        }
+      },
+      "properties": {
+        "cacheSizeGB": 3072,
+        "scalingFactor": 1.0,
+        "subnet": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/scgroup/providers/Microsoft.Network/virtualNetworks/scvnet/subnets/sub1",
+        "encryptionSettings": {
+          "keyEncryptionKey": {
+            "keyUrl": "https://keyvault-cmk.vault.azure.net/keys/key2047/test",
+            "sourceVault": {
+              "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/scgroup/providers/Microsoft.KeyVault/vaults/keyvault-cmk"
+            }
+          }
+        },
+        "directoryServicesSettings": {
+          "activeDirectory": {
+            "primaryDnsIpAddress": "192.0.2.10",
+            "secondaryDnsIpAddress": "192.0.2.11",
+            "cacheNetBiosName": "contosoSmb",
+            "domainName": "contosoAd.contoso.local",
+            "domainNetBiosName": "contosoAd",
+            "credentials": {
+              "username": "consotoAdmin",
+              "password": "<password>"
+            }
```

Comment:
I have some worries about accepting/storing active directory passwords. Isn't there any more secure alternative than password authentication we could use?

Reply:
Copied from a separate email thread on this topic.

The AD credentials are required so that the AOS storage cache cluster can communicate with AD.

We take care to not log these fields or output anything containing these fields while forwarding them to the AOS storage cache cluster that requires them.
We have suppress logs flags in the code as well to make sure the code path does not log them.

Between the time of the PUT and when we send them to the cluster once it is built, we temporarily store the credentials in a key vault.
Once the credentials are sent to AOS we remove them from the keyvault.